service/service.go

```diff
@@ -289,6 +289,22 @@ func (s *Service) handleClientMsg(msg ws.Message) error {
 		s.connMap[sessionID] = msg.ConnID
 		s.mut.Unlock()
 
+		return nil
+	case ClientMessageReconnect:
+		data, ok := cm.Data.(map[string]string)
+		if !ok {
+			return fmt.Errorf("unexpected data type: %T", cm.Data)
+		}
+		sessionID := data["sessionID"]
+		if sessionID == "" {
+			return fmt.Errorf("missing sessionID in client message")
+		}
+
+		s.log.Debug("reconnect message, updating connMap", mlog.String("sessionID", sessionID))
+		s.mut.Lock()
+		s.connMap[sessionID] = msg.ConnID
```

Comment:
I know we're doing it above too, but is there any chance we won't have a ConnID, and should check for it?

Reply:
The `ConnID` comes straight from the websocket server so it must be set.
Are you asking about the case in which a reconnect occurs but no previous session is in the map?

Reply:
The first -- I haven't looked at the server websocket code. :)

Cool, thanks!